src/components/Article/ToggleArticleEngagement.tsx

```diff
@@ -18,24 +19,67 @@ export function ToggleArticleEngagement({
   const currentUser = useCurrentUser();
   const queryUtils = trpc.useContext();
   const { data } = trpc.user.getArticleEngagement.useQuery(undefined, { enabled: !!currentUser });
+  const { data: bookmarkedArticles } = trpc.user.getBookmarkedArticles.useQuery(undefined, {
+    enabled: !!currentUser,
+  });
 
   const isToggled = useMemo(() => {
     if (!data) return undefined;
     const keys = Object.keys(data) as ArticleEngagementType[];
-    return keys.reduce<Partial<Record<ArticleEngagementType, boolean>>>((acc, key) => {
+    const engagements = keys.reduce<Partial<Record<ArticleEngagementType, boolean>>>((acc, key) => {
       const ids = data[key] ?? [];
       return { ...acc, [key]: !!ids.find((id) => id === articleId) };
```

Comment:
I believe this is relatively faster than using `.find()`
```suggestion
      return { ...acc, [key]: !!ids.includes(articleId) };
```